### PR TITLE
Iv 142

### LINKED ISSUE #605: Benchmark thea_staking Pallet

### PR DESCRIPTION
**Pull request by 35359595**
## Describe your changes
Testing and benchmarks for thea slashing

`nominate()` is rendered useless after modifications to `IndividualExposure` to make it always have relayer nominating after `bond()` call - that's why tests are cleaned up. this change is to make sure stakers can not bail mid-session from slash scheduled relayers to avoid been slashed.

## Issue ticket number and link
#142 from Thea
Closes #605 

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

### CODE REVIEW

File: pallets/thea-staking/src/benchmarking.rs [hunk 1]
```diff
@@ -0,0 +1,251 @@
+// This file is part of Polkadex.
+
+// Copyright (C) 2020-2022 Polkadex oü.
```
Comment (serhii-temchenko) [inline pallets/thea-staking/src/benchmarking.rs]: Please update to 2023

File: pallets/thea-staking/src/tests.rs [hunk 2]
```diff
@@ -113,7 +115,7 @@ fn test_nominate_with_valid_arguments_returns_ok() {
 		register_nominator();
 		let (candidate, network_id, bls_key) = get_candidate();
 		let nominator = 2;
-		assert_ok!(TheaStaking::nominate(Origin::signed(nominator), candidate));
+		// assert_ok!(TheaStaking::nominate(Origin::signed(nominator), candidate));
```
Comment (serhii-temchenko) [inline pallets/thea-staking/src/tests.rs]: Please remove

File: pallets/thea-staking/src/tests.rs [hunk 3]
```diff
@@ -153,24 +155,30 @@ fn test_nominate_with_already_staked_relayer_returns_staker_already_nominating()
...
 			Error::<Test>::StakerAlreadyNominating
 		);
 	})
 }
 
+// as now binding happens instantly without option this test now tests
+// that re-nominating is not happening on second nomination
 #[test]
-fn test_nominate_with_wrong_candidate_returns_candidate_not_found() {
+fn test_nominate_with_wrong_candidate_returns_candidate_alread_nominated() {
```
Comment (serhii-temchenko) [inline pallets/thea-staking/src/tests.rs]: Typo in `alread`

File: pallets/thea-staking/src/tests.rs [hunk 4]
```diff
@@ -203,7 +211,7 @@ fn test_unbond_with_valid_arguments_returns_ok() {
 		register_nominator();
 		let (candidate, network, bls_key) = get_candidate();
 		let nominator = 2;
-		assert_ok!(TheaStaking::nominate(Origin::signed(nominator), candidate));
+		// assert_ok!(TheaStaking::nominate(Origin::signed(nominator), candidate));
```
Comment (serhii-temchenko) [inline pallets/thea-staking/src/tests.rs]: Same

File: pallets/thea-staking/src/tests.rs [hunk 5]
```diff
@@ -243,21 +251,28 @@ fn test_unbond_with_zero_nomination_returns_ok() {
...
-			backing: None,
+			backing: 1,
 			unlocking: vec![unlocking_chunk],
 		};
 		assert_eq!(TheaStaking::stakers(nominator), Some(nominator_exposure));
 	})
 }
 
+// use thea_staking::session::{Exposure, IndividualExposure, StakingLimits, UnlockChunk};
+
```
Comment (serhii-temchenko) [inline pallets/thea-staking/src/tests.rs]: Same

File: pallets/thea-staking/src/tests.rs [hunk 6]
```diff
@@ -243,21 +251,28 @@ fn test_unbond_with_zero_nomination_returns_ok() {
...
 #[test]
 fn test_withdraw_unbounded_with_returns_ok() {
 	new_test_ext().execute_with(|| {
 		register_candidate();
 		insert_staking_limit();
 		register_nominator();
 		unbonding();
+		// panic!("here");
```
Comment (serhii-temchenko) [inline pallets/thea-staking/src/tests.rs]: Same

File: pallets/thea-staking/src/tests.rs [hunk 7]
```diff
@@ -480,7 +495,7 @@ fn test_unbond_with_amount_equal_to_staked_amount_returns_ok() {
 		let candidate = 1u64;
 		let network_id = 0;
 		let bls_key = BLSPublicKey([1; 192]);
-		assert_ok!(TheaStaking::nominate(Origin::signed(nominator), candidate));
+		// assert_ok!(TheaStaking::nominate(Origin::signed(nominator), candidate));
```
Comment (serhii-temchenko) [inline pallets/thea-staking/src/tests.rs]: Same

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/thea-staking/src/benchmarking.rs (hunk 1 above)
Comment (35359595) [reply pallets/thea-staking/src/benchmarking.rs]: done

File: pallets/thea-staking/src/tests.rs (hunk 2 above)
Comment (35359595) [reply pallets/thea-staking/src/tests.rs]: done

File: pallets/thea-staking/src/tests.rs (hunk 3 above)
Comment (35359595) [reply pallets/thea-staking/src/tests.rs]: done

File: pallets/thea-staking/src/tests.rs (hunk 5 above)
Comment (35359595) [reply pallets/thea-staking/src/tests.rs]: done

File: pallets/thea-staking/src/tests.rs (hunk 4 above)
Comment (35359595) [reply pallets/thea-staking/src/tests.rs]: done

File: pallets/thea-staking/src/tests.rs (hunk 6 above)
Comment (35359595) [reply pallets/thea-staking/src/tests.rs]: done

File: pallets/thea-staking/src/tests.rs (hunk 7 above)
Comment (35359595) [reply pallets/thea-staking/src/tests.rs]: done

**Review by serhii-temchenko — APPROVED** [review]

**Review by felixfaisal — APPROVED** [review]
LGTM

**Pull request merged**